doc/refman/RefMan-ssr.tex

```diff
@@ -0,0 +1,4703 @@
+generalize or
+rewrite such a definition; on the other hand, it is automatically
+expanded during
+computation, whereas expanding the equation \ssrC{En} requires explicit
+rewriting.
+
+The use of this equation name generation option with a \ssrC{case} or an
+\ssrC{elim} tactic changes the status of the first \iitem{}, in order to
+deal with the possible parameters of the constants introduced.
+
+On the
+goal \ssrC{a <> b} where \ssrC{a, b} are natural numbers, the tactic:
+\begin{lstlisting}
+  case E : a => [|n].
+\end{lstlisting}
+generates two subgoals. The equation \ssrC{E : a = 0}  (resp. \ssrC{E : a =
+  S n}, and the constant \ssrC{n : nat}) has been added to
+the context of the goal \ssrC{0 <> b} (resp. \ssrC{S n  <> b}).
+
+If the user does not provide a branching \iitem{} as first \iitem{},
+or if the \iitem{} does not provide enough names for the arguments of
+a constructor,
+then the constants generated are introduced under fresh \ssr{} names.
+For instance, on the goal \ssrC{a <> b}, the tactic:
+\begin{lstlisting}
+  case E : a => H.
+\end{lstlisting}
+also generates two subgoals, both requiring a proof of \ssrC{False}.
+ The hypotheses \ssrC{E : a = 0} and
+\ssrC{H : 0 = b} (resp. \ssrC{E : a = S _n_} and
+\ssrC{H : S _n_ = b}) have been added to the context of the first
+subgoal (resp. the second subgoal).
+
+Combining the generation of named equations mechanism with the
+\ssrC{case} tactic strengthens the power of a case analysis. On the other
+hand, when combined with the \ssrC{elim} tactic, this feature is mostly
+useful for
+debug purposes, to trace the values of decomposed parameters and
+pinpoint failing branches.
+
+% This feature is also useful
+% to analyse and debug generate-and-test style scripts that prove program
+% properties by generating a large set of input patterns and uniformly
+% solving the corresponding subgoals by computation and rewriting, e.g,
+
+% \begin{lstlisting}
+%   case: et => [|e' et]; first by case: s.
+%   case: e => //; case: b; case: w.
+% \end{lstlisting}
+% If the above sequence fails, then it's easy enough to replace the line
+% above with
+% \begin{lstlisting}
+%   case: et => [|e' et].
+%   case Ds: s; case De: e => //; case Db: b; case Dw: w=> [|s' w'] //=.
+% \end{lstlisting}
+% Then the first subgoal that appears will be the failing one, and the
+% equations \ssrC{Ds}, \ssrC{De}, \ssrC{Db}
+% and \ssrC{Dw} will pinpoint its branch. When the constructors of
+% the decomposed type have arguments (like \ssrC{w : (seq nat)}
+% above) these need to be
+% introduced in order to generate the equation, so there should
+% always be an explicit \iitem{} (\ssrC{\[|s' w'\]} above) that
+% assigns names to these arguments. If this \iitem{}
+% is omitted the arguments are introduced with default
+% name; this
+% feature should be
+% avoided except for quick debugging runs (it has some uses in complex tactic
+% sequences, however).
+
+
+
+\subsection{T
```

Comment:
I'm wondering whether it makes sense to keep this information. What is the version of SSR distributed with Coq? 1.6.1? 1.6.2? 1.7? Is this information displayed anywhere? It isn't anymore when loading `ssreflect` in coqtop.